### PR TITLE
add assistance module first-commit

### PR DESCRIPTION
**Pull request by Joaquin-Huaranga**
add page new page and route assistance
image
add acls
image
buttons to mark assistance (Se deshabilitan dependiendo de si has marcado asistencia o salida)
image
type entry and outlet
image
image

### CODE REVIEW

File: packages/hosting/src/pages/assistance/GetAssistance.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      isEntry={isEntry}
+    />
+  );
+};
+
+const AssistanceButtons = ({ handleMarkAssistance, isEntry }) => {
+  return (
+    <Row gutter={[16, 16]}>
+      <Col span={24}>
+        <Flex>
```
Comment (adminAgenciaServitec) [inline packages/hosting/src/pages/assistance/GetAssistance.jsx:53]: Agregar un gap:5

File: packages/hosting/src/firebase/collections/assistance.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export const getAssistancesId = () => assistancesRef.doc().id;
+
+export const fetchAssistance = async (id) =>
+  fetchDocumentOnce(assistancesRef.doc(id));
+
+export const fetchAssistances = async () =>
+  fetchCollectionOnce(assistancesRef.where("isDeleted", "==", false));
+
+export const fetchAssistancesByUserID = async (userId) =>
```
Comment (adminAgenciaServitec) [inline packages/hosting/src/firebase/collections/assistance.js:15]: fix fetchAssistancesByUserID to fetchAssistancesByUserId

File: packages/hosting/src/pages/assistance/GetAssistance.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    const assistanceData = {
+      id: assistanceId || getAssistancesId(),
+      userName: userFullName(user),
+      type: type,
+      date: dayjs(currentDate).format(DATE_FORMAT_TO_FIRESTORE),
+    };
+
+    try {
+      await addAssistance(assistanceData);
```
Comment (adminAgenciaServitec) [inline packages/hosting/src/pages/assistance/GetAssistance.jsx:28]: Falta el assignCreated de deafultFirestore

**Review by adminAgenciaServitec — CHANGES_REQUESTED** [review]

File: packages/hosting/src/pages/assistance/GetAssistance.jsx (hunk 3 above)
Comment (Joaquin-Huaranga) [reply packages/hosting/src/pages/assistance/GetAssistance.jsx]: image

File: packages/hosting/src/pages/assistance/index.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  const { authUser } = useAuthentication();
+  const { assignDeleteProps } = useDefaultFirestoreProps();
+
+  const [assistances = [], assistancesLoading, assistancesError] =
+    useCollectionData(assistancesRef.where("isDeleted", "==", false));
+
+  useEffect(() => {
+    assistancesError && notification({ type: "error" });
+  }, [assistancesError]);
+
```
Comment (adminAgenciaServitec) [inline packages/hosting/src/pages/assistance/index.jsx:19]: if(assistancesLoading) return <Spinner height="80svh"/>

**Review by adminAgenciaServitec — CHANGES_REQUESTED** [review]

File: packages/hosting/src/pages/assistance/index.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  if (assistancesLoading) {
+    return;
```
Comment (adminAgenciaServitec) [inline packages/hosting/src/pages/assistance/index.jsx:21]:
```ts 
 if (assistancesLoading) return <Spinner height="80svh"/>;
```

**Review by adminAgenciaServitec — CHANGES_REQUESTED** [review]

**Review by adminAgenciaServitec — APPROVED** [review]

**Pull request merged**